sphinxmixcrypto/node.py

```diff
@@ -82,17 +62,59 @@ def get_dimensions(self):
 
 
 @attr.s(frozen=True)
-class SphinxPacket(object):
+class SphinxHeader(object):
     """
-    I am a decoded sphinx packet
+    The Sphinx header.
+
+    The Sphinx paper refers to the header fields as the greek letters: alpha, beta and gamma.
     """
     alpha = attr.ib(validator=attr.validators.instance_of(bytes))
     beta = attr.ib(validator=attr.validators.instance_of(bytes))
     gamma = attr.ib(validator=attr.validators.instance_of(bytes))
+
+
+@attr.s(frozen=True)
+class SphinxBody(object):
+    """
+    A Sphinx has the body of a lion or lioness.  The sphinx packet
+    body is repeated encrypted with the lioness wide-block cipher.
+
+    The Sphinx paper refers to this field of the packet as the greek letter delta.
+    """
     delta = attr.ib(validator=attr.validators.instance_of(bytes))
 
 
 @attr.s(frozen=True)
+class SphinxPacket(object):
+    """
+    I am a decoded sphinx packet
+    """
+    header = attr.ib(validator=attr.validators.instance_of(SphinxHeader))
+    body = attr.ib(validator=attr.validators.instance_of(SphinxBody))
+
+    def get_raw_bytes(self):
+        """
+        Get all the bytes.
+        """
+        return self.header.alpha + self.header.beta + \
+            self.header.gamma + self.body.delta
+
+    @classmethod
+    def from_raw_bytes(cls, params, raw_packet):
+        """
+        Create a SphinxPacket given the raw bytes and
+        an instance of SphinxParams.
+        """
+        assert isinstance(params, SphinxParams)
+        alpha, beta, gamma, delta = params.get_dimensions()
+        _alpha = raw_packet[:alpha]
+        _beta = raw_packet[alpha:alpha + beta]
+        _gamma = raw_packet[alpha + beta:alpha + beta + gamma]
+        _delta = raw_packet[alpha + beta + gamma:]
```

Comment:
(optional)

Hmmm `alpha` and `_alpha`, etc.  What about factoring the slicing out so you could do something like `alpha, beta, gamma, delta = params.get_bytes(raw_packet)`?